### PR TITLE
Consume Automation Logger & Add rule to dump MSAL logs to its own file 

### PR DESCRIPTION
**Pull request by shahzaibj**
Depends on https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1138

The file be named like `test_name-msal.log`

### CODE REVIEW

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/MsalLoggingRule.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                    case INFO:
+                        msalfileLogAppender.append(LogLevel.INFO, tag, message, null);
+                        break;
+                    case ERROR:
+                        msalfileLogAppender.append(LogLevel.ERROR, tag, message, null);
+                        break;
+                    case WARNING:
+                        msalfileLogAppender.append(LogLevel.WARN, tag, message, null);
+                        break;
+                }
```
Comment (AdamBJohnsonx) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/MsalLoggingRule.java:86]: nit: if you converted the level first, or used a function to do it, there would only be one call to msialfilelogappender.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/MsalLoggingRule.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                        msalLogFileAppender.getLogFile(),
+                        LOG_FOLDER_NAME
+                );
+            }
+        };
+    }
+
+    private FileAppender turnOnMsalLogging(final Description description) throws IOException {
+        final String msalLogFileName = description.getMethodName() + "-msal.log";
+        final FileAppender msalfileLogAppender = new FileAppender(msalLogFileName, new LogcatLikeFormatter());
```
Comment (AdamBJohnsonx) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/MsalLoggingRule.java:67]: nit: msalFileLogAppender?

**Review by AdamBJohnsonx — APPROVED** [review]

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/MsalLoggingRule.java (hunk 1 above)
Comment (shahzaibj) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/MsalLoggingRule.java]: Addressed

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/MsalLoggingRule.java (hunk 2 above)
Comment (shahzaibj) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/MsalLoggingRule.java]: Addressed

**Pull request merged**